### PR TITLE
Schema additions for Zip support

### PR DESCRIPTION
**Pull request by ryfu-msft**
This PR make additions to the 1.3 schema to support ZIP installers.

Changes:
Addition of:
- NestedInstallerType
- NestedInstallerFiles
-- RelativeFilePath
-- PortableCommandAlias

Existing test manifests that used `zip` installerType have now been replaced with `msi` in order to avoid schema validation errors that are specific to `zip`.

Tests:
- Manifest with zip installer with multiple nested installers
- Manifest with zip installer with no nested installer file
- Manifest with zip installer with no nested installer type
- Manifest with zip installer with no RelativeFilePath defined
- v1.3 manifest with zip installers

Related to:
#2270 
#140

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-cli/pull/2288)

### CODE REVIEW

File: schemas/JSON/manifests/v1.3.0/manifest.installer.1.3.0.json [hunk 1]
```diff
@@ -69,6 +69,45 @@
       ],
       "description": "Enumeration of supported installer types. InstallerType is required in either root level or individual Installer level"
     },
+    "NestedInstallerType": {
+      "type": [ "string", "null" ],
```
Comment (yao-msft) [inline schemas/JSON/manifests/v1.3.0/manifest.installer.1.3.0.json:73]: >[ "string", "null" ], [](http://example.com/codeflow?start=14&length=21)

This is required, so "string" here

File: schemas/JSON/manifests/v1.3.0/manifest.singleton.1.3.0.json [hunk 2]
```diff
@@ -111,6 +111,45 @@
...
+        "portable"
+      ],
+      "description": "Enumeration of supported nested installer types contained inside an archive file"
+    },
+    "NestedInstallerFiles": {
+      "type": [ "array", "null" ],
+      "items": {
+        "type": "object",
+        "title": "NestedInstallerFile",
+        "properties": {
```
Comment (yao-msft) [inline schemas/JSON/manifests/v1.3.0/manifest.singleton.1.3.0.json:134]: > "properties" [](http://example.com/codeflow?start=7&length=13)

Update the types here as well

**Review by yao-msft — APPROVED** [review]

File: src/AppInstallerCommonCore/Manifest/ManifestValidation.cpp [hunk 3]
```diff
@@ -175,6 +175,30 @@ namespace AppInstaller::Manifest
                 }
             }
 
+            if (installer.InstallerType == InstallerTypeEnum::Zip)
```
Comment (JohnMcPMS) [inline src/AppInstallerCommonCore/Manifest/ManifestValidation.cpp:178]: Would be better as:
```suggestion
            if (IsArchiveType(installer.InstallerType))
```
to make the intention more clear for the future.

File: src/AppInstallerCommonCore/Manifest/ManifestYamlPopulator.cpp [hunk 4]
```diff
@@ -821,6 +876,19 @@ namespace AppInstaller::Manifest
                 installer.AppsAndFeaturesEntries = manifest.DefaultInstallerInfo.AppsAndFeaturesEntries;
             }
 
+            if (installer.InstallerType == InstallerTypeEnum::Zip)
```
Comment (JohnMcPMS) [inline src/AppInstallerCommonCore/Manifest/ManifestYamlPopulator.cpp:879]: Use semantic `IsArchiveType` function here as well.

**Review by JohnMcPMS — CHANGES_REQUESTED** [review]
Using the semantic function will help future maintainers understand what was actually being looked for.

File: src/AppInstallerCommonCore/Manifest/ManifestValidation.cpp (hunk 3 above)
Comment (ryfu-msft) [reply src/AppInstallerCommonCore/Manifest/ManifestValidation.cpp]: Changed to suggested

File: src/AppInstallerCommonCore/Manifest/ManifestYamlPopulator.cpp (hunk 4 above)
Comment (ryfu-msft) [reply src/AppInstallerCommonCore/Manifest/ManifestYamlPopulator.cpp]: Changed to using `IsArchiveType` function

**Review by JohnMcPMS — APPROVED** [review]

**Pull request merged**